python/cudf/cudf/_lib/pylibcudf/groupby.pyx

```diff
@@ -211,9 +230,12 @@ cdef class GroupBy:
             A tuple whose first element is the group's keys and whose second
             element is a table of values with nulls replaced.
         """
-        cdef pair[unique_ptr[table], unique_ptr[table]] c_res = move(
-            dereference(self.c_obj).replace_nulls(value.view(), replace_policies)
-        )
+        cdef pair[unique_ptr[table], unique_ptr[table]] c_res
+        cdef vector[replace_policy] c_replace_policies = replace_policies
```

Comment:
Just curious: is the change here needed because we can't rely on cython to reliably convert a `List[replace_policy]` to a `vector[replace_policy]`? Or was there some sort of other bug in the prior code?

Reply:
It's because the conversion of a Python object requires the GIL, so we have to do it outside the `with nogil` block :disappointed: For our extension types (Column/Table) I made it work by reacquiring the GIL internally in the `view` methods while marking the methods `nogil` so that we could write the code more briefly, but I don't think we can do that for these conversions.